echopype/calibrate/env_params.py

```diff
@@ -59,199 +68,289 @@ def harmonize_env_param_time(
+        raise ValueError("'formula_absorption' has to be None, 'FG' or 'AM' for EK echosounders.")
+    if (
+        out_dict["formula_sound_speed"] is not None
+        and out_dict["formula_sound_speed"] != "Mackenzie"
+    ):
+        raise ValueError("'formula_absorption' has to be None or 'Mackenzie' for EK echosounders.")
+
+    # Calculation sound speed and absorption requires at least T, S, P
+    # tsp_all_exist controls wherher to calculate sound speed and absorption
+    tspa_all_exist = np.all(
+        [out_dict[p] is not None for p in ["temperature", "salinity", "pressure", "pH"]]
+    )
 
-    # Initiate input/output dict
-    out_dict = {}
-    if user_env_dict is None:
-        user_env_dict = {}
-
-    # Re-calculate environment parameters if user supply all env variables
-    tsp_all_exist = np.all([p in user_env_dict for p in ["temperature", "salinity", "pressure"]])
-
-    if tsp_all_exist:
-        out_dict["sound_speed"] = uwa.calc_sound_speed(
-            temperature=user_env_dict["temperature"],
-            salinity=user_env_dict["salinity"],
-            pressure=user_env_dict["pressure"],
-        )
-        out_dict["sound_absorption"] = uwa.calc_absorption(
-            frequency=freq,
-            temperature=user_env_dict["temperature"],
-            salinity=user_env_dict["salinity"],
-            pressure=user_env_dict["pressure"],
-        )
-    # Otherwise
-    #  get temperature, salinity, and pressure from raw data file
-    #  get sound speed from user inputs or raw data file
-    #  get absorption from user inputs or computing from env params stored in raw data file
-    else:
-        # pressure is encoded as "depth" in EK80  # TODO: change depth to pressure in EK80 file?
+    # If EK80, get env parameters from data if not provided in user dict
+    # All T, S, P, pH are needed because we always have to compute sound absorption for EK80 data
+    if not tspa_all_exist and sonar_type == "EK80":
         for p_user, p_data in zip(
-            ["temperature", "salinity", "pressure", "pH", "sound_speed"],
-            ["temperature", "salinity", "depth", "acidity", "sound_speed_indicative"],
+            ["temperature", "salinity", "pressure", "pH"],  # name in defined env params
+            ["temperature", "salinity", "depth", "acidity"],  # name in EK80 data
         ):
-            out_dict[p_user] = user_env_dict.get(p_user, echodata["Environment"][p_data])
+            out_dict[p_user] = user_dict.get(p_user, env[p_data])
+
+    # Sound speed
+    if out_dict["sound_speed"] is None:
+        if not tspa_all_exist:
+            # sounds speed always exist in EK60 and EK80 data
+            out_dict["sound_speed"] = env["sound_speed_indicative"]
+            out_dict.pop("formula_sound_speed")
+        else:
+            # default to Mackenzie sound speed formula if not in user dict
+            if out_dict["formula_sound_speed"] is None:
+                out_dict["formula_sound_speed"] = "Mackenzie"
```

Comment:
see my suggestion and comment above
```suggestion
            if out_dict["formula_sound_speed"] in (None, "Mackenzie"):
```

Reply:
I added this but this was actually to set the `formula_sound_speed` to Mackenzie when it does not exist, not an if statement like anove. I've reverted it back.